app/components/createContainer/createContainerController.js

```diff
@@ -231,6 +241,13 @@ function ($scope, $state, $stateParams, $filter, Config, Info, Container, Contai
       networkMode += ':' + containerName;
     }
     config.HostConfig.NetworkMode = networkMode;
+    var extraHosts = [];
```

Comment:
Instead of creating a temporary `extraHosts` variable, just map onto the `config.HostConfig.ExtraHosts` each entry that has a value.

E.g.

```
    config.HostConfig.ExtraHosts.map(function (v) {
      if (v.value) {
        return v.value;
      }
    });
```